language-plutus-core/src/Language/PlutusCore/TypeSynthesis.hs

```diff
@@ -72,20 +72,22 @@ kindOfTypeBuiltin TySize       = sizeToType
 -- | Annotate a 'Type'. Invariant: the type must be in normal form. The invariant is not checked.
 -- In case a type is open, an 'OpenTypeOfBuiltin' is returned.
 -- We use this for annotating types of built-ins (both static and dynamic).
-annotateClosedNormalType
-    :: (AsTypeError e a, MonadError e m)
+annotateNormalizeType
+    :: (AsTypeError e a, MonadError e m, MonadQuote m)
     => a -> Builtin () -> Type TyName () -> m (NormalizedType TyNameWithKind ())
-annotateClosedNormalType ann con ty = case annotateType ty of
+annotateNormalizeType ann con ty = case annotateType ty of
     Left  (_::RenameError ()) -> throwing _TypeError $ InternalTypeErrorE ann $ OpenTypeOfBuiltin ty con
-    Right annTyOfName         -> pure $ NormalizedType annTyOfName
+    -- That's quite inefficient, but is there anything we can do about that?
+    Right annTyOfName         -> normalizeType annTyOfName
```

Comment:
We want to prenormalize types of dynamic built-in names. Previously they were normalized, but now we have dynamic types which can be non-normalized. That's a future work.